app/uk/gov/hmrc/agentsubscription/auth/AuthActions.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription.auth
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.mvc._
+import uk.gov.hmrc.agentsubscription.connectors.AuthConnector
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+trait AuthActions {
+  me: Results =>
+
+  val authConnector: AuthConnector
+
+  protected val agentWithEnrolments = new ActionBuilder[RequestWithEnrolments] with ActionRefiner[Request, RequestWithEnrolments] {
```

Comment:
also this is used both by SubscriptionController which needs to know what the enrolments are and RegistrationController which doesn't. If this just made the enrolments URL available instead of the actual enrolments then an HTTP request would be saved when RegistrationController is invoked.